app/plugins/Wallet/js/lifecycle.js

```diff
@@ -1,89 +1,163 @@
+	});
+	return;
+
+	/* Fetch all wallet transactions by iterate over wallet addresses
+	var loopmax = result.addresses.length;
+	var counter = 0;
+	(function next() {
+		setTimeout(function() {
+			updateAddrTxn(result.addresses[counter].address);
+			next();
+		}, 50); // force 50 ms delay between each GET request
+	})();*/
+});
+
+// Get transactions for a specific wallet address
+function updateAddrTxn(addr) {
+	eID('transaction-list').innerHTML = '';
+	IPC.sendToHost('api-call', {
+		url: '/wallet/transactions/' + addr,
 		type: 'GET',
-		args: {
-			startheight: 0,
-			// arbitrarily large endheight to get full history
-			endheight: 1000000
-		}
 	}, 'update-history');
-	*/
-	
-	updating = setTimeout(update, 2500);
 }
 
-// Update transaction history and addresses
-addResultListener('update-status', function(result) {
-	wallet = result;
-	
+// Update wallet summary in header
+addResultListener('update-status', function(wallet) {
 	// Update balance confirmed and uncomfirmed
 	var bal = convertSiacoin(wallet.confirmedsiacoinbalance);
 	var pend = convertSiacoin(wallet.unconfirmedincomingsiacoins).sub(convertSiacoin(wallet.unconfirmedoutgoingsiacoins));
 	if (wallet.unlocked && wallet.encrypted) {
+		eID('confirmed').style.display = 'inline';
+		eID('unconfirmed').style.display = 'inline';
 		eID('confirmed').innerHTML = 'Balance: ' + bal + ' S';
-		eID('uncomfirmed').innerHTML = 'Pending: ' + pend + ' S';
-		eID('confirmed').style.removeProperty('display');
-		eID('uncomfirmed').style.removeProperty('display');
+		eID('unconfirmed').innerHTML = 'Pending: ' + pend + ' S';
 	} else {
-		eID('confirmed').style.setProperty('display', 'none');
-		eID('uncomfirmed').style.display = 'none';
+		eID('confirmed').style.display = 'none';
+		eID('unconfirmed').style.display = 'none';
 	}
 });
 
-/*
-// Adds an address to the address list
+// Append wallet address to address list
+function appendAddress(addr) {
+	// Create only new addresses
+	if (typeof(addr) == 'undefined') { return; }
+	var addrElement = eID('addressbp').cloneNode(true);
+
+	// DOM shortcut
+	var field = function(selector) {
+		return addrElement.querySelector(selector);
+	};
+
+	// Insert values
+	field('.listnum').innerHTML = eID('address-list').childNodes.length + 1;
+	field('.address').innerHTML = addr.address;
+	field('.address').id = addr.address;
+	field('.address').addEventListener("click", getAddress);
+
+	// Display address
+	eID('address-list').appendChild(addrElement);
+	show(addrElement);
+	return;
+}
+function getAddress(event) {
+	updateAddrTxn(event.target.id);
+}
+
 function appendTransaction(txn) {
-	if (eID(txn.transactionid)) {
-		return;
+	if (typeof(txn) == 'undefined') { return; }
+
+	// Add only new transactions
+	if (eID(txn.transactionid)) { return; }
+	var txnElement = eID('transactionbp').cloneNode(true);
+	txnElement.id = txn.transactionid;
+	txnElement.timestamp = txn.confirmationtimestamp * 1000;
+
+	// DOM shortcut
+	var field = function(selector) {
```

Comment:
Oh this was another possible lag-point. I'm not sure if defining this function for temporary usage in `appendTransaction` opens up the possibility of a memleak or something.